src/pybind/mgr/dashboard/controllers/docs.py

```diff
@@ -14,21 +14,56 @@ class Docs(BaseController):
 
     @classmethod
     def _gen_tags(cls, all_endpoints):
-        ctrl_names = set()
+        """ Generates a list of all tags and corresponding descriptions. """
+        # Scenarios to consider:
+        #     * Intentionally make up a new tag name at controller => New tag name displayed (as intended).
+        #     * Misspell or make up a new tag name at endpoint => Tag not displayed. Endpoint not displayed.
+        #     * Misspell tag name at controller (when refering to another controller) => Tag displayed but no endpoints assigned
+        #     * Description for a certain tag added at multiple locations => Only one description is displayed
+        # #TODO: OK?
+        list_of_ctrl = set()
         for endpoints in ENDPOINT_MAP.values():
             for endpoint in endpoints:
                 if endpoint.is_api or all_endpoints:
-                    ctrl_names.add(endpoint.group)
+                    list_of_ctrl.add(endpoint.ctrl)
+
+        tags = []
+        for ctrl in list_of_ctrl:
+            tag_name = ctrl.__name__
+            tag_descr = "*No description available*"
+
+            if hasattr(ctrl, '_doc_info'):
+                if ctrl._doc_info['tag']:
+                    tag_name = ctrl._doc_info['tag']
+                if ctrl._doc_info['tag_descr']:
+                    tag_descr = ctrl._doc_info['tag_descr']
+            tags.append({
```

Comment:
This will not work when two controllers are using the same "group" but only one of them specifies the "description". You should use a map for the tags variable instead of a list so that in the end you generate a single tag with the correct description.

Reply:
Thanks, will change!